### PR TITLE
Bug 1154993 - [flame][3.0]Can't view SMS that has no phone number associ...

### PR DESCRIPTION
**Pull request by julienw**
...ated r=azasypkin

We should never break because of an API error.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=08ae81ceca1fcef7364cecbb433fb1c1de4d9e20

File: apps/sms/test/unit/contacts_test.js [hunk 1]
```diff
@@ -527,10 +527,10 @@ suite('Contacts', function(done) {
     });
 
     test('The mozContacts find() call returned an error', function(done) {
-      Contacts.findByPhoneNumber('callonerror').then(
-        () => done(new Error('The promise should be rejected.')),
-        () => done()
-      );
+      Contacts.findByPhoneNumber('callonerror').then((results) => {
```
Comment (azasypkin) [inline apps/sms/test/unit/contacts_test.js]: Quite severe regression actually and we both didn't manage to catch it during review :) How do you feel about simple integration test for this bug?

Comment (julienw) [inline apps/sms/test/unit/contacts_test.js]: why not but I feel we need your other patch that has not landed yet and I think this patch needs to go in asap. Do you think it's good enough if an integration tests is done in a follow-up patch?

Comment (azasypkin) [inline apps/sms/test/unit/contacts_test.js]: Yep, follow-up sounds good to me - replied the same at bugzilla :)

**Pull request merged**